### PR TITLE
Feature/login layout

### PR DESCRIPTION
**Pull request by tcurran13**
This branch is good to go boys.

### CODE REVIEW

File: app/src/main/java/com/example/qrgo/SignupActivity.java [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+        //Set up variables
+        final String TAG = "Sample";
+        Button mRegister;
+        final EditText username;
+        final EditText email;
+        final EditText phone;
+        FirebaseFirestore db;
+
+        mRegister = findViewById(R.id.register);
+        username = findViewById(R.id.username);
```
Comment (fardeenfs) [inline app/src/main/java/com/example/qrgo/SignupActivity.java:70]: Username and the user's name are different. The username is autogenerated for every user, when they sign up. You have to create a function for this on sign up. Use a different variable name for the user's name here (i.e. userFirstName and userLastName).

File: app/src/main/java/com/example/qrgo/SignupActivity.java [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+                //Snapshot listener function potentially  needing implementation
+                collectionUsers.addSnapshotListener(new EventListener<QuerySnapshot>() {
+                    @Override
+                    public void onEvent(@Nullable QuerySnapshot value, @Nullable FirebaseFirestoreException error) {
+                        //Prime programming real estate
+                    }
+                });
+                username.setText("");
+                email.setText("");
+                phone.setText("");
```
Comment (fardeenfs) [inline app/src/main/java/com/example/qrgo/SignupActivity.java]: Do not use FirebaseFirestore from here.

Please use the following functions on the FirebaseConnect class:

- checkImei()
- addNewUser()
- addNewPlayerProfile()

These functions are made in accordance with our agreed upon schema. You can check the comments on FirebaseConnect, to see the required params.

File: app/src/main/res/layout/activity_signup.xml [hunk 3]
```diff
@@ -0,0 +1,156 @@
...
+        android:layout_marginTop="10dp"
+        android:layout_marginEnd="33dp"
+        android:layout_marginRight="33dp"
+        android:text="Name"
+        android:textColor="@color/black"
+        android:textSize="15dp"
+        android:textStyle="bold" />
+
+    <EditText
+        android:id="@+id/username"
```
Comment (fardeenfs) [inline app/src/main/res/layout/activity_signup.xml:46]: Username and the user's name are different. Change the id of this element to something else to prevent confusion, (eg. edit_name)

File: app/src/main/res/values/strings.xml [hunk 4]
```diff
@@ -1,3 +1,4 @@
 <resources>
     <string name="app_name">QRgo</string>
+    <string name="register" />
```
Comment (fardeenfs) [inline app/src/main/res/values/strings.xml]: This line is not required.

File: app/src/main/res/values/ids.xml [hunk 5]
```diff
@@ -0,0 +1,11 @@
+<?xml version="1.0" encoding="utf-8"?>
+<resources>
+    <item name="signup" type="id" />
+    <item name="name" type="id" />
+    <item name="username" type="id" />
+    <item name="email" type="id" />
+    <item name="address" type="id" />
+    <item name="phone" type="id" />
+    <item name="phone_number" type="id" />
+    <item name="register" type="id" />
```
Comment (fardeenfs) [inline app/src/main/res/values/ids.xml]: This page is not required.

File: build.gradle [hunk 6]
```diff
@@ -16,8 +16,8 @@ buildscript {
     }
 }
 plugins {
-    id 'com.android.application' version '7.4.0' apply false
-    id 'com.android.library' version '7.4.0' apply false
```
Comment (fardeenfs) [inline build.gradle]: Please revert these changes. Do not update the gradle version.

**Review by lucasr4200 — APPROVED** [review]

**Review by 1pharaxh — APPROVED** [review]

File: app/src/main/java/com/example/qrgo/SignupActivity.java [hunk 7]
```diff
@@ -0,0 +1,128 @@
...
+
+                            }
+                        });
+
+                        db.addNewPlayerProfile(userName, firstName, lastName, contactEmail, contactPhone, 0, 0, 0, new FirebaseConnect.OnUserProfileAddListener() {
+                            @Override
+                            public void onUserProfileAdd(boolean success) {
+
+                            }
+                        });
```
Comment (fardeenfs) [inline app/src/main/java/com/example/qrgo/SignupActivity.java]: Please call db.addNewPlayerProfile() inside the onUserAdd(), so that you wait for the user to be added before adding their profile.

File: app/src/main/java/com/example/qrgo/SignupActivity.java [hunk 8]
```diff
@@ -0,0 +1,128 @@
...
+                }
+
+                // Clear fields on signup page
+                username.setText("");
+                email.setText("");
+                phone.setText("");
+
+                // Navigate to Home Activity
+                Intent intent = new Intent(SignupActivity.this, HomeActivity.class);
+                startActivity(intent);
```
Comment (fardeenfs) [inline app/src/main/java/com/example/qrgo/SignupActivity.java]: Please put these inside the onUserProfileAdd(), so that you wait for the profile to be added before redirecting them to the Home Page. Otherwise the Home Page might send a read request before the write operation is complete.

File: app/src/main/java/com/example/qrgo/SignupActivity.java [hunk 9]
```diff
@@ -0,0 +1,128 @@
...
+        username = findViewById(R.id.username);
+        email = findViewById(R.id.address);
+        phone = findViewById(R.id.phone_number);
+
+
+
+
+
+        mRegister.setOnClickListener(new View.OnClickListener() {
+            @Override
```
Comment (fardeenfs) [inline app/src/main/java/com/example/qrgo/SignupActivity.java:79]: Disable the registration button, to prevent multiple signups

File: app/src/main/java/com/example/qrgo/SignupActivity.java [hunk 10]
```diff
@@ -0,0 +1,128 @@
...
+                final String realName = username.getText().toString();
+
+
+
+                // Traverse the string
+                for (int i = 0; i < realName.length(); i++) {
+                    if(realName.charAt(i) == ' '){
+                        final String firstName = realName.substring(0,i);
+                        final String lastName = realName.substring(i+1);
+                        final String userName = lastName.charAt(0)+firstName;
```
Comment (fardeenfs) [inline app/src/main/java/com/example/qrgo/SignupActivity.java:96]: This works. However, we have to check if this is unique (multiple users can have the same name). We will ignore this for now as there is no FirebaseConnect function for this yet. Just add a //TODO Username Check above this line, so that we do not forget.

**Comment by fardeenfs** [comment]
Perfect! Let's merge this :)

**Review by fardeenfs — APPROVED** [review]
